File: backend/src/main/java/com/stockflow/backend/entity/StockMovement.java
package com.stockflow.backend.entity;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import org.hibernate.annotations.CreationTimestamp;

import java.time.LocalDateTime;

@Entity
@Table(name = "stock_movement")
@AllArgsConstructor
@NoArgsConstructor
@Getter @Setter
public class StockMovement {

    public enum StockMovementType {
        COMPRA,
        AJUSTE_POSITIVO,
        AJUSTE_NEGATIVO,
        VENTA
    }

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private StockMovementType type;

    @Column(nullable = false)
    private int quantity;

    @CreationTimestamp
    @Column(nullable = false, updatable = false)
    private LocalDateTime date;

    @ManyToOne
    @JoinColumn(name = "product_id", nullable = false)
    private Product product;

    @ManyToOne
    @JoinColumn(name = "user_id", nullable = false)
    private User user;
}

File: frontend/src/types/stockMovement.ts
import type { ProductResponse } from './product';

export type StockMovementType = 'COMPRA' | 'AJUSTE_POSITIVO' | 'AJUSTE_NEGATIVO' | 'VENTA';

export interface StockMovementResponse {
  id: number;
  type: StockMovementType;
  quantity: number;
  date: string;
  product: ProductResponse;
  userId: number;
  userName: string;
}